### ISSUE TITLE
When the pages are deleted, the overview doesn't reflect the change.

### ISSUE THREAD

**Issue by omarNaifer12** [docmost/docmost#581]
![General-Docmost-Personal-MicrosoftEdge2024-12-2223-51-21-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/1ef4b720-1938-49a1-9656-860ba2bd4673)

**Comment by Philipinho** [comment]
Will have to invalidate the space `recent-changes` query key on page deletion.
https://github.com/docmost/docmost/blob/290b7d9d9418a82affbecd2172bf98db61138f7f/apps/client/src/features/page/queries/page-query.ts#L137-L145

**Comment by omarNaifer12** [comment]
Yes, that’s it. I’m wondering which approach has better performance: simply calling the API again or handling it manually? For example, we could store the ID of the deleted page in an atom and use useEffect to track when it’s not null. Then, we could use recursion to delete all its children. It seems a bit complex. What do you think?
